module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,58 @@
+#include "compression.hpp"
+#include <algorithm>
+#include <iostream>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& imgToCompress) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressedImg;
```

Comment:
Reserve memory, you can predict size. And then call `shirnk_to_fit`

Reply:
@nauka-programowania-MA  the reason to do that is memory allocation efficiency or some other? I mean, if I reserve the memory at the beginning as you said, then I don't need to resize vector later so the code will be faster? That's it?

Reply:
Yes. And you can predict size. You have width and height. So in worse scenario you will need all to reserved memory, and if not you will free it when call `shrink_to_fit`. This method you will avoid few memory allocation.